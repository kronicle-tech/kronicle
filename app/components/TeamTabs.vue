<template>
  <div class="my-3">
    <b-button class="my-1" :to="`/teams/${teamId}`">Info</b-button>
    <b-button class="my-1" :to="`/teams/${teamId}/dependencies`"
      >Visualizations</b-button
    >
    <b-button class="my-1" :to="`/teams/${teamId}/components`"
      >Components</b-button
    >
    <b-button
      class="my-1"
      :to="`/teams/${teamId}/cross-functional-requirements`"
    >
      Cross Functional Requirements
    </b-button>
    <b-button class="my-1" :to="`/teams/${teamId}/tech-debts`"
      >Tech Debts</b-button
    >
    <b-button class="my-1" :to="`/teams/${teamId}/tests`">Tests</b-button>
    <b-button class="my-1" :to="`/teams/${teamId}/errors`">Errors</b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
  props: {
    teamId: {
      type: String,
      required: true,
    },
  },
})
</script>
