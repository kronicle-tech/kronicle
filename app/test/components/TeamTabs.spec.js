import { mount } from '@vue/test-utils'
import TeamTabs from '@/components/TeamTabs.vue'

describe('TeamTabs', () => {
  let propsData
  let wrapper
  const createWrapper = () => {
    wrapper = mount(TeamTabs, { propsData })
  }

  beforeEach(() => {
    propsData = {
      teamId: 'test-team-id',
    }
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  test('renders the component', () => {
    createWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
