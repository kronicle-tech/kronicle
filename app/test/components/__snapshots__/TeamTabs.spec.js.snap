// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamTabs renders the component 1`] = `
"<div class=\\"my-3\\"><a href=\\"/teams/test-team-id\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\"> Info </a> <a href=\\"/teams/test-team-id/diagrams\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Diagrams
  </a> <a href=\\"/teams/test-team-id/docs\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\"> Docs </a> <a href=\\"/teams/test-team-id/components\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Components
  </a> <a href=\\"/teams/test-team-id/cross-functional-requirements\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Cross Functional Requirements
  </a> <a href=\\"/teams/test-team-id/tech-debts\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Tech Debts
  </a> <a href=\\"/teams/test-team-id/openapi-specs\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    OpenAPI
  </a> <a href=\\"/teams/test-team-id/graphql-schemas\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    GraphQL
  </a> <a href=\\"/teams/test-team-id/tests\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\"> Tests </a> <a href=\\"/teams/test-team-id/errors\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\"> Errors </a></div>"
`;
